tests/unit/CoreListTest.php

```diff
@@ -0,0 +1,753 @@
+			$this->markTestSkipped( 'This test requires WP 6.2 or higher.' );
+		}
+
+		$block_content = '
+			<!-- wp:list -->
+				<ul class="wp-block-list">
+					<!-- wp:list-item {"style":{"typography":{"textDecoration":"underline"}}} -->
+						<li style="text-decoration:underline">List Item 2</li>
+					<!-- /wp:list-item -->
+				</ul>
+			<!-- /wp:list -->
+		';
+
+		// Set post content.
+		wp_update_post(
+			[
+				'ID'           => $this->post_id,
+				'post_content' => $block_content,
+			]
+		);
+
+		$query     = '
+			fragment CoreListBlockFragment on CoreList {
+				attributes {
+					anchor
+					backgroundColor
+					className
+					cssClassName
+					fontFamily
+					fontSize
+					gradient
+					lock
+					# metadata
+					ordered
+					placeholder
+					reversed
+					start
+					style
+					textColor
+					type
+					values
+				}
+			}
+
+			fragment CoreListItemBlockFragment on CoreListItem {
+				attributes {
+					className
+					content
+					fontFamily
+					fontSize
+					lock
+					# metadata
+					placeholder
+					style
+				}
+			}
+
+			query Post( $id: ID! ) {
+				post(id: $id, idType: DATABASE_ID) {
+					databaseId
+					editorBlocks( flat: false ) {
+						apiVersion
+						blockEditorCategoryName
+						clientId
+						cssClassNames
+						innerBlocks {
+							apiVersion
+							blockEditorCategoryName
+							clientId
+							cssClassNames
+							name
+							parentClientId
+							renderedHtml
+							... on CoreListItem {
+								...CoreListItemBlockFragment
+							}
+						}
+						isDynamic
+						name
+						parentClientId
+						renderedHtml
+						...CoreListBlockFragment
+					}
+				}
+			}
+		';
+		$variables = [
+			'id' => $this->post_id,
+		];
+
+		$actual = graphql( compact( 'query', 'variables' ) );
+
+		$this->assertArrayNotHasKey( 'errors', $actual, 'There should not be any errors' );
+		$this->assertArrayHasKey( 'data', $actual, 'The data key should be present' );
+		$this->assertArrayHasKey( 'post', $actual['data'], 'The post key should be present' );
+
+		$this->assertEquals( $this->post_id, $actual['data']['post']['databaseId'], 'The post ID should match' );
+
+		$this->assertEquals( 1, count( $actual['data']['post']['editorBlocks'] ) );
+
+		$block = $actual['data']['post']['editorBlocks'][0]['innerBlocks'][0];
+
+		$this->assertNotEmpty( $block['apiVersion'], 'The apiVersion should be present' );
+		$this->assertEquals( 'text', $block['blockEditorCategoryName'], 'The blockEditorCategoryName should be text' );
+		$this->assertEmpty( $block['cssClassNames'], 'The cssClassNames should be present' );
+		$this->assertNotEmpty( $block['clientId'], 'The clientId should be present' );
+		$this->assertEquals( 'core/list-item', $block['name'], 'The block name should be core/list' );
+
+		// @todo : The 'parentClientId' attribute is NULL in the response.
+		// $this->assertNotEmpty( $block['parentClientId'], 'There should be some parentClientId for the block' );
```

Comment:
With `flat: true `, we should be able to uncomment this and `assertEquals()` to the `$actual['data']['post']['editorBlocks'][0]['clientId']`.